src/Microsoft.PowerShell.ConsoleHost/host/msh/ProgressNode.cs

```diff
@@ -336,6 +356,83 @@ private static void RenderFullDescription(string description, string indent, int
                     maxWidth));
         }
 
+        /// <summary>
+        /// Renders a node in the "ANSI" style.
+        /// </summary>
+        /// <param name="strCollection">
+        /// List of strings to which the node's rendering will be appended.
+        /// </param>
+        /// <param name="indentation">
+        /// The indentation level in chars at which the node should be rendered.
+        /// </param>
+        /// <param name="maxWidth">
+        /// The maximum number of chars that the rendering is allowed to consume.
+        /// </param>
+        private
+        void
+        RenderAnsi(ArrayList strCollection, int indentation, int maxWidth)
+        {
+            string indent = StringUtil.Padding(indentation);
+            string secRemain = string.Empty;
+            if (SecondsRemaining >= 0)
+            {
+                secRemain = SecondsRemaining.ToString() + "s";
+            }
+
+            int secRemainLength = secRemain.Length + 1;
+
+            // limit progress bar to 120 chars as no need to render full width
+            if (PSStyle.Instance.Progress.MaxWidth > 0 && maxWidth > PSStyle.Instance.Progress.MaxWidth)
+            {
+                maxWidth = PSStyle.Instance.Progress.MaxWidth;
```

Comment:
I notice that `PSStyle.Instance.Progress` is referred to a lot.

As it is, it's quite a verbose invocation and at least a theoretical Law of Demeter violation, but more importantly invites a race condition. This code depends on a consistent progress style setting, but the singleton instance could be changed by another thread at any point in this code.

Instead, if you assign the style to a local at the top of the method, you eliminate at least the first two issues. If you get a copy of the progress style at that point, you eliminate the third too.

Reply:
Thinking on it more I don't think the first two points are that compelling. It's readable as it is. (Although there are good reasons for the Law of Demeter).

But we should work through the fact that we conceptually should be locking the singleton when reading from it.